WTrackTableView: don't use header section highlight…

ronso0 opened the pull request:
...explicitely style with qss only.

As soon as the tableview is focused all column labels would be automatically rendered **bold** -- and previous style is not restored when the tableview is unfocused, only when the table content changes. This can not be overwritten with qss, so it can screw up the skin design.
Also, due to selectionModel()->selectedRows() it is not even useful to highlight the focused column.

I did not find a qss way to override the hader font for focused table views -- doing this in c++ is the last resort (and I have no idea why it would be enabled in the first place because I don't see UI benefit).
No side effects to be expected https://doc.qt.io/qt-5/qheaderview.html#highlightSections-prop

- [ ] double check skins after #3788 is merged

Holzhaus commented:
Looks good I think.

![Screenshot from 2021-05-03 23-03-56](https://user-images.githubusercontent.com/1834516/116933689-2e64d800-ac64-11eb-9f4a-6301a6157263.png)
![Screenshot from 2021-05-03 23-03-26](https://user-images.githubusercontent.com/1834516/116933691-2f960500-ac64-11eb-932c-2f8dd73f1133.png)
![Screenshot from 2021-05-03 23-03-01](https://user-images.githubusercontent.com/1834516/116933692-2f960500-ac64-11eb-9260-2ca73cee4953.png)
![Screenshot from 2021-05-03 23-02-19](https://user-images.githubusercontent.com/1834516/116933695-302e9b80-ac64-11eb-8994-847acbeb5cea.png)

Ready for merge?

ronso0 commented:
not really. the reason for this PR is that I use Open Sans Semibold for the header, but `setHighlightSections(true);` would make it bold. I rebased onto 2.3 and LateNight and it should look like this:
![image](https://user-images.githubusercontent.com/5934199/116938883-b13d6100-ac6b-11eb-9c8b-b2920a69153f.png)

Other skins are using bold anyway.

ronso0 commented:
@Holzhaus Can you check what fonts are recognized on your machine?
I checked with Windows 10 and the font changes in #3788 fixed the font issues there.
this is what I see:
```
 <!-- this exact same block is printed three times in a row -->
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Condensed Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Semibold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "SemiBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Extrabold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Extrabold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Condensed Light" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Regular" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Semibold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
...
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
...
```

Holzhaus commented:
```
Debug [Main]: Loading resources from  "/home/jan/Projects/mixxx/res/"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Regular" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "SemiBold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "SemiBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Regular" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "SemiBold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "SemiBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Regular" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "SemiBold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "SemiBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Bold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "ExtraBold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Open Sans" style: "Semibold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Bold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Light" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Medium Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Regular" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Bold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Medium" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Bold" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Light Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Light" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Medium Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Regular" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Bold Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Italic" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Debug [Main]: FONT LOADED family: "Ubuntu" style: "Medium" point sizes: "6,7,8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72"
Info [Main]: DbConnection - Available drivers for database connections: ("QIBASE", "QSQLITE", "QMARIADB", "QMYSQL", "QMYSQL3", "QODBC", "QODBC3", "QPSQL", "QPSQL7", "QTDS", "QTDS7")
```

Holzhaus commented:
Headers are still bold:
![Screenshot from 2021-05-04 12-54-01](https://user-images.githubusercontent.com/1834516/116993707-20e93580-acd8-11eb-9b8e-038fb52ee25e.png)

ronso0 committed:
WTrackTableView: don't use header section highlight, explicitely style with qss only

ronso0 commented:
We know Qt versions make a difference so I assume it's an qss issue again.

To be 100% sure the c++ change is in place I explicitely `setHighlightSections(false)`, rebased and force-pushed.
I also adjusted the qss to explicitely include `QHeaderView::section`. Maybe that makes a difference for you.

For me, the original issue with bold headers is resolved, the headers are semi-bold, both in the focused and unfocused table:
![image](https://user-images.githubusercontent.com/5934199/117029840-56ede000-acff-11eb-99d3-5e83b9b7de0a.png)

Ready for merge.

Meanwhile I'll build the Qt 5.15.2 sources and see if I can debug the remaining font & icon issues.

Holzhaus commented:
I think it works now, thanks.

![Screenshot from 2021-05-04 20-25-25](https://user-images.githubusercontent.com/1834516/117051787-39c50b80-ad17-11eb-97e1-0c622a6e727b.png)
